tests/queries/0_stateless/02403_enable_extended_results_for_datetime_functions.sql.j2

```diff
@@ -0,0 +1,8 @@
+{% for option_value in ['true', 'false'] -%}
+{% for date_fun in ['toStartOfYear', 'toStartOfISOYear', 'toStartOfQuarter', 'toStartOfMonth', 'toStartOfWeek', 'toMonday', 'toLastDayOfMonth'] -%}
+SELECT {{ date_fun }}(toDate32('1950-02-02')) SETTINGS enable_extended_results_for_datetime_functions = {{ option_value }};
```

Comment:
Maybe it's paranoid but since the extended day types were recently extended from 1900 to 2299 (1925 to 2283 before), it could make sense to use the "new" date range (1900-1934, 2284-2299) in the test.

Reply:
I put 1920 date instead of 1950 for tests. I found several issues with date LUT arithmetic in about 5 date/time functions. I think I will create a separate PR to fix it and create new functional tests specially to check the boundaries. 

Also, I made j2 output more clear. It outputs now:
toStartOfYear;toDate32;true	1920-01-01
type;toStartOfYear;toDate32;true	Date32
toStartOfYear;toDateTime64;true	1920-01-01
type;toStartOfYear;toDateTime64;true	Date32

Also based on tests I found that narrowing was not implemented for toStartOfISOYear so I corrected it.